### PR TITLE
Enable environment variables with clang_format_path.

### PR DESCRIPTION
**Pull request by sean-mcmanus**
For https://github.com/Microsoft/vscode-cpptools/issues/2344 .

### CODE REVIEW

File: Extension/package.json [hunk 1]
```diff
@@ -71,7 +71,7 @@
             "null"
           ],
           "default": null,
-          "description": "The full path of the clang-format executable.",
+          "description": "The full path of the clang-format executable. Changing this setting requires a Reload Window if it contains environment variables.",
```
Comment (bobbrow) [inline Extension/package.json]: did it not work to intercept the didChangeConfigurations message in protocallFilter.ts?

**Review by bobbrow — CHANGES_REQUESTED** [review]

File: Extension/package.json (hunk 1 above)
Comment (sean-mcmanus) [reply Extension/package.json]: I didn't investigate that. I can.

Comment (sean-mcmanus) [reply Extension/package.json]: It doesn't appear possible to "replace" a setting in that filter, but I can change it to ignore that particular setting change and send our own message with the correct setting.

Comment (bobbrow) [reply Extension/package.json]: Yes, I think that would be better.  Thanks!

File: Extension/src/LanguageServer/client.ts [hunk 2]
```diff
@@ -402,6 +403,10 @@ class DefaultClient implements Client {
             if (changedSettings["commentContinuationPatterns"]) {
                 updateLanguageConfigurations();
             }
+            if (changedSettings["clang_format_path"]) {
+                let settings: CppSettings = new CppSettings(this.RootUri);
+                this.languageClient.sendNotification(UpdateClangFormatPathNotification, util.resolveVariables(settings.clangFormatPath, null));
```
Comment (bobbrow) [inline Extension/src/LanguageServer/client.ts:408]: Can you just use `changedSettings["clang_format_path"]` without creating a new `CppSettings` object?

**Review by bobbrow — CHANGES_REQUESTED** [review]

File: Extension/src/LanguageServer/client.ts (hunk 2 above)
Comment (sean-mcmanus) [reply Extension/src/LanguageServer/client.ts:408]: No, the value is "...".

Comment (bobbrow) [reply Extension/src/LanguageServer/client.ts:408]: 😆 ok

**Review by bobbrow — APPROVED** [review]

**Pull request merged**